catalog/app/website/pages/AwsMarketplace/Form.tsx

```diff
@@ -0,0 +1,310 @@
+      }) => (
+        <form className={cx(classes.root, className)} onSubmit={handleSubmit}>
+          <M.Typography className={classes.description}>
+            Quilt is available in the AWS Marketplace. We bring seamless collaboration to
+            S3 by connecting people, pipelines, and machines using visual, verifiable,
+            versioned data packages.
+          </M.Typography>
+          <div className={classes.group}>
+            <RF.Field
+              InputProps={{ classes: inputClasses }}
+              className={classes.input}
+              component={TextField}
+              name="firstName"
+              placeholder="First Name*"
+              disabled={submitting}
+              validate={validators.required as FF.FieldValidator<any>}
+              errors={{
+                required: 'First name is required',
+              }}
+            />
+            <RF.Field
+              InputProps={{ classes: inputClasses }}
+              className={classes.input}
+              component={TextField}
+              name="lastName"
+              placeholder="Last Name*"
+              disabled={submitting}
+              validate={validators.required as FF.FieldValidator<any>}
+              errors={{
+                required: 'Last name is required',
+              }}
+            />
+          </div>
+          <div className={classes.group}>
+            <RF.Field
+              InputProps={{ classes: inputClasses }}
+              className={classes.input}
+              component={TextField}
+              name="companyName"
+              placeholder="Company Name*"
+              disabled={submitting}
+              validate={validators.required as FF.FieldValidator<any>}
+              errors={{
+                required: 'Company name is required',
+              }}
+            />
+          </div>
+          <div className={classes.group}>
+            <RF.Field
+              InputProps={{ classes: inputClasses }}
+              className={classes.input}
+              component={TextField}
+              name="companyEmail"
+              placeholder="Company Email*"
+              disabled={submitting}
+              validate={validators.required as FF.FieldValidator<any>}
+              errors={{
+                required: 'Company email is required',
+              }}
+            />
+          </div>
+          <SubmitStatus
+            submitting={submitting}
+            error={error || submitError}
+            success={submitSucceeded}
+          />
+          <M.Typography className={classes.note}>
+            By submitting this form, I agree to receive email updates about Quilt
+          </M.Typography>
+          <div className={classes.actions}>
+            <M.Button
+              type="submit"
+              color="secondary"
+              variant="contained"
+              disabled={submitting || (submitFailed && hasValidationErrors)}
+            >
+              Submit
```

Comment:
"submit" (and "tanks for submission") seems too abstract. we should use a more descriptive language here. tbh i don't understand what this form does at all. like, what am i "submitting"? subscribing to the mailing list? requesting a sales call? the UX is quite confusing